### ISSUE TITLE
getting non-segmentation -> segmentation traingin and evaluation working

### ISSUE THREAD

**Issue by morngrar** [morngrar/isic-study#1]
The main script now is `train.py`. This is run by the docker container on a VM that has nvidia  virtual GPU passtrough. So the repo is ran with `docker-compose up --build` and then the train script is run by the `run.sh` script.

Introducing the segmentation code  as a step between two model fits fails with C type erros  or numpy cardinality erros depending on the sequence I implement them.

The intended result is to store the result of the model without segmentation and with segmentation on the same augmented data set, and repeat for differently generated data sets for a uni report.

I got everything working --- albeit with a terrible accuracy for the model, but I get confusing stack traces only when trying to include teh segmentation step (which worked fine locally in an isolated case, btw)

**Comment by jimmyroberts** [comment]
Hi, jimraro here from twitch. Is it required to have Nvidia on your machine? I'm trying to run it on macOS and I get a runtime error with docker. Please let me know if there is an alternative and I'll proceed with assistance. :) Cheers

` => => naming to docker.io/library/isic-study_mlmodel                      0.0s
[+] Running 0/0
 ⠋ Container isic-study-mlmodel-1  Cre...                                  0.0s
Error response from daemon: Unknown runtime specified nvidia
`

**Comment by morngrar** [comment]
Hi! 

Well, it's not a requirement per se, but the VM has cuda as the only available backend for tensorflow, at least afaik, so that's what I work with, and it's what the Dockerfile is set up for. But the code itself is only using keras and tensorflow besides some python packages on top. See Dockerfile for deps. I'm able on my laptop to run the code without docker on CPU only. In that case run the shell script mentioned inreadme and this issue.

**Comment by morngrar** [comment]
Turns out to run two fits consecutively one has to delete the existing model and manually run the garbage collector to free the GPU memory. That was what was acting up, ultimately. But the error mesages weren't obvious.
